cli/flamingock-cli/src/main/java/io/flamingock/cli/factory/SqlDataSourceFactory.java

```diff
@@ -0,0 +1,110 @@
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+public class SqlDataSourceFactory {
+
+    public static DataSource createSqlDataSource(DatabaseConfig.SqlConfig config) {
+        if (config == null) {
+            throw new IllegalArgumentException("SQL configuration is required");
+        }
+
+        if (config.getEndpoint() == null) {
+            throw new IllegalArgumentException("JdbcUrl is required");
+        }
+
+        if (config.getUsername() == null) {
+            throw new IllegalArgumentException("Database username is required");
+        }
+
+        if (config.getPassword() == null) {
+            throw new IllegalArgumentException("Database password is required");
+        }
+
+        if (config.getSqlDialect() == null) {
+            throw new IllegalArgumentException("Sql dialect is required");
+        }
+
+        try {
+            DataSource sqlDatasource;
+
+            if (config.getSqlDialect().equals(SqlDialect.SQLITE)) {
+                SQLiteDataSource sqliteDatasource = new SQLiteDataSource();
+                sqliteDatasource.setUrl(config.getEndpoint());
+
+                try (Connection conn = sqliteDatasource.getConnection();
+                     Statement stmt = conn.createStatement()) {
+                    stmt.execute("PRAGMA journal_mode=WAL;");
+                    stmt.execute("PRAGMA busy_timeout=5000;");
+                }
+
+                sqlDatasource = sqliteDatasource;
+            } else {
+                HikariConfig datasourceConfig = new HikariConfig();
+                datasourceConfig.setDriverClassName(config.getDriverClassName());
+
+                if (config.getSqlDialect().equals(SqlDialect.H2)) {
+                    datasourceConfig.setJdbcUrl("jdbc:h2:mem:testdb;DB_CLOSE_DELAY=-1");
+                    datasourceConfig.setUsername("sa");
+                    datasourceConfig.setPassword("");
+                } else {
+                    datasourceConfig.setJdbcUrl(config.getEndpoint());
+                    datasourceConfig.setUsername(config.getUsername());
+                    datasourceConfig.setPassword(config.getPassword());
+                }
+
+                if (config.getSqlDialect() == SqlDialect.INFORMIX) {
+                    datasourceConfig.setMaximumPoolSize(1);
+                    datasourceConfig.setMinimumIdle(0);
+                    datasourceConfig.setConnectionTimeout(30000);
+                    datasourceConfig.setIdleTimeout(600000);
+                    datasourceConfig.setMaxLifetime(1800000);
+                    datasourceConfig.setLeakDetectionThreshold(0);
+                    datasourceConfig.setValidationTimeout(5000);
+                    datasourceConfig.setConnectionTestQuery("SELECT 1 FROM systables WHERE tabid=1");
+                    datasourceConfig.setInitializationFailTimeout(-1);
+                    datasourceConfig.setAutoCommit(true);
```

Comment:
We ned to document why these values are hardcoded and why we picked these ones.
This needs to be done in the documentation. Meanwhile, we can write down somewhere, in the ticket itself in trello, for example